apis/sgv2-quarkus-common/src/test/java/io/stargate/sgv2/common/CqlEnabledIntegrationTestBase.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package io.stargate.sgv2.common;
+
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.config.DriverConfigLoader;
+import com.datastax.oss.driver.api.core.config.OptionsMap;
+import com.datastax.oss.driver.api.core.config.TypedDriverOption;
+import com.datastax.oss.driver.internal.core.loadbalancing.DcInferringLoadBalancingPolicy;
+import java.time.Duration;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.TestInstance;
+
+/**
+ * Base test class that exposes a Java driver {@link CqlSession} connected to the Stargate
+ * coordinator, allowing subclasses to create and populate their CQL schema.
+ *
+ * <p>Subclasses must be annotated with {@link io.quarkus.test.junit.QuarkusIntegrationTest}.
+ */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public abstract class CqlEnabledIntegrationTestBase {
```

Comment:
cool stuff, can be used in docs api as well.. Do you think in future having `CqlSession` like stuff as annotation would make sense? Or this is enough?

Reply:
This can probably be turned into a test resource, now that the contact point info is passed as system properties. The only requirement is that `StargateTestResource` needs to execute first, do you know if there is a way to enforce a reliable order?

Reply:
Ok apparently there is `QuarkusTestResourceLifecycleManager.order()`, I'll give it a try.

Reply:
Hrmm actually a test resource might not be the best fit:

> These resources are started before the first test is run, and are closed at the end of the test suite.

There's no class-level lifecycle that would allow us to create the session and a dedicated keyspace before each test.

Reply:
Yea I was more thinking about Junit5 extension..

Reply:
Follow-up ticket: #2054